nixos/modules/services/monitoring/prometheus/exporters/script.nix

```diff
@@ -0,0 +1,46 @@
+{ config, lib, pkgs, options }:
+
+with lib;
+
+let
+  cfg = config.services.prometheus.exporters.script;
+  configFile = pkgs.writeText "script-exporter.yaml" (builtins.toJSON cfg.settings);
+in
+{
+  port = 9172;
+  extraOpts = {
+    settings = mkOption {
+      type = types.attrs;
```

Comment:
I only took a brief look at the manual, but it should be possible to harden the type to use e.g. `attrsOf (listOf (str))` (or maybe `attrsOf (listOf (anything))`

Reply:
Good idea, I tried to encode the whole config format in nix.